src/base/bittorrent/session.cpp

```diff
@@ -2982,6 +3020,14 @@ bool Session::hasPerTorrentRatioLimit() const
     return false;
 }
 
+bool Session::hasPerTorrentSeedingTimeLimit() const
+{
+    foreach (TorrentHandle *const torrent, m_torrents)
+        if (torrent->seedingTimeLimit() >= 0) return true;
+
```

Comment:
Why it is redundant?
It doesn't break any coding style rules here, but just separates loop.